specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+  /// trusted CA certificates.
+  [propget] HRESULT MutuallyTrustedCertificates([out, retval]
+      ICoreWebView2ClientCertificateList** clientCertificateList);
+
+  /// Returns the selected certificate.
+  [propget] HRESULT SelectedCertificate([out, retval]
+      ICoreWebView2ClientCertificate** certificate);
+
+  /// Sets the certificate to respond to the server.
+  [propput] HRESULT SelectedCertificate(
+      [in] ICoreWebView2ClientCertificate* certificate);
+
+  /// The state of the Client Certificate Request Response State.
+  /// See `COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE` for descriptions of states.
+  /// The default is COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITH_DEFAULT_DIALOG.
+  [propget] HRESULT ClientCertificateRequestResponseState(
+      [out, retval] COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE* state);
+
+  /// Sets the `Client Certificate Request Response State` property.
+  [propput] HRESULT ClientCertificateRequestResponseState(
+      [in] COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE state);
+
+  /// Returns an `ICoreWebView2Deferral` object.  Use this operation to
+  /// complete the event at a later time.
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+}
+```
+
+## .NET/ WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2ClientCertificateRequestedEventArgs;
+    runtimeclass CoreWebView2ClientCertificate;
+
+    enum CoreWebView2ClientCertificateRequestResponseState
+    {
+        AbortRequest = 0,
+        ContinueWithoutCertificate = 1,
+        ContinueWithCertificate = 2,
+        ContinueWithDefaultDialog = 3,
+    };
+    enum CoreWebView2ClientCertificateKind
+    {
+        SmartCard = 0,
+        Pin = 1,
+        Other = 2,
+    };
+
+    runtimeclass CoreWebView2ClientCertificateRequestedEventArgs
+    {
+        // ICoreWebView2ClientCertificateRequestedEventArgs members
+        String Host { get; };
+        Int32 Port { get; };
+        Boolean IsProxy { get; };
+        IVector<string> CertificateAuthorities { get; };
+        IVector<CoreWebView2ClientCertificate> MutuallyTrustedCertificates { get; };
+        CoreWebView2ClientCertificate SelectedCertificate { get; set; };
+        CoreWebView2ClientCertificateRequestResponseState ClientCertificateRequestResponseState { get; set; };
+
+        Windows.Foundation.Deferral GetDeferral();
+    }
+
+    runtimeclass CoreWebView2ClientCertificate
+    {
+        // ICoreWebView2ClientCertificate members
+        String Subject { get; };
+        String Issuer { get; };
+        Windows.Foundation.DateTime ValidStartDate { get; };
+        Windows.Foundation.DateTime ValidExpiryDate { get; };
+        String SerialNumber { get; };
+        String DisplayName { get; };
+        String PEMEncodedData { get; };
+        IVector<string> PEMEncodedIssuerChainData { get; };
```

Comment:
.NET uses just a list of strings, whereas C++ uses a special interface named after the fact that it's a list of chain data.

Should the C++ interface just be a `ICoreWebView2StringList` that we reuse for anybody who produces a list of strings?

Reply:
Yes let's do that.